lnrpc/signrpc/signer_server.go

```diff
@@ -547,3 +567,34 @@ func (s *Server) DeriveSharedKey(_ context.Context, in *SharedKeyRequest) (
 
 	return &SharedKeyResponse{SharedKey: sharedKeyHash[:]}, nil
 }
+
+// parseRawKeyBytes checks that the provided raw public key is valid and returns
+// the public key.
+func parseRawKeyBytes(rawKeyBytes []byte) (*btcec.PublicKey, error) {
+	var (
+		targetPubKey *btcec.PublicKey
+		err          error
+	)
+
+	switch {
+	// If the user provided a raw key, but it's of the
+	// wrong length, then we'll return with an error.
+	case len(rawKeyBytes) != 0 && len(rawKeyBytes) != 33:
+
+		return nil, fmt.Errorf("pubkey must be " +
+			"serialized in compressed format if " +
+			"specified")
+
+	// If a proper raw key was provided, then we'll attempt
+	// to decode and parse it.
+	case len(rawKeyBytes) != 0 && len(rawKeyBytes) == 33:
+		targetPubKey, err = btcec.ParsePubKey(
```

Comment:
Could just be `return btcec.ParsePubKey(rawKeyBytes, btcec.S256())`. The `ParsePubKey` already has nice error messages, no need to wrap it again.
Also, then we could get rid of the two `var`s at the top of the function and return an explicit `return nil, nil` at the end of the function to make it clearer what happens.

Reply:
fixed, much better.